module.go

```diff
@@ -1,137 +1,188 @@
-	objdet "go.viam.com/rdk/vision/objectdetection"
-	"go.viam.com/rdk/vision/viscapture"
-	"go.viam.com/utils/rpc"
+	svision "go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+	vision "go.viam.com/rdk/vision"
 )
 
-var (
-	ObstaclesDistance = resource.NewModel("viam", "obstacles-distance", "obstacles-distance")
-	errUnimplemented  = errors.New("unimplemented")
-)
+var ObstaclesDistance = resource.NewModel("viam", "vision", "obstacles-distance")
 
-func init() {
-	resource.RegisterService(vision.API, ObstaclesDistance,
-		resource.Registration[vision.Service, *Config]{
-			Constructor: newObstaclesDistanceObstaclesDistance,
-		},
-	)
-}
+// DefaultNumQueries is the default number of times the camera should be queried before averaging.
+const DefaultNumQueries = 10
 
-type Config struct {
-	/*
-		Put config attributes here. There should be public/exported fields
-		with a `json` parameter at the end of each attribute.
-
-		Example config struct:
-			type Config struct {
-				Pin   string `json:"pin"`
-				Board string `json:"board"`
-				MinDeg *float64 `json:"min_angle_deg,omitempty"`
-			}
-
-		If your model does not need a config, replace *Config in the init
-		function with resource.NoNativeConfig
-	*/
+// DistanceDetectorConfig specifies the parameters for the camera to be used
+// for the obstacle distance detection service.
+type DistanceDetectorConfig struct {
+	NumQueries    int    `json:"num_queries"`
+	DefaultCamera string `json:"camera_name"`
 }
 
-// Validate ensures all parts of the config are valid and important fields exist.
-// Returns implicit required (first return) and optional (second return) dependencies based on the config.
-// The path is the JSON path in your robot's config (not the `Config` struct) to the
-// resource being validated; e.g. "components.0".
-func (cfg *Config) Validate(path string) ([]string, []string, error) {
-	// Add config validation code here
-	return nil, nil, nil
+func init() {
+	resource.RegisterService(svision.API, ObstaclesDistance, resource.Registration[svision.Service, *DistanceDetectorConfig]{
+		Constructor: func(
+			ctx context.Context, deps resource.Dependencies, c resource.Config, logger logging.Logger,
+		) (svision.Service, error) {
+			attrs, err := resource.NativeConfig[*DistanceDetectorConfig](c)
+			if err != nil {
+				return nil, err
+			}
+			return registerObstacleDistanceDetector(ctx, c.ResourceName(), attrs, deps, logger)
+		},
+	})
 }
 
-type obstaclesDistanceObstaclesDistance struct {
-	resource.AlwaysRebuild
+// Validate ensures all parts of the config are valid.
+func (config *DistanceDetectorConfig) Validate(path string) ([]string, []string, error) {
+	deps := []string{}
+	var optionalDeps []string
+	if config.DefaultCamera == "" {
+		return nil, optionalDeps, errors.Errorf(`expected "camera_name" attribute (DefaultCamera) for obstacles pointcloud at %q`, path)
+	}
+	deps = append(deps, config.DefaultCamera)
```

Comment:
In the original version, the default camera was optional. Instead of lines 54-57,

```go
if config.DefaultCamera != "" {
    deps = append(deps, config.DefaultCamera)
}
```

Reply:
Recording what we figured out IRL: if the default camera is the empty string and you don't specify a nondefault camera in your GRPC request, you get errors about not knowing which camera to use. If the default is empty and you _do_ specify a nondefault camera in your request, you can get errors about not being able to get the camera because it's not in your dependencies. If you're not going to use the default camera, you need to explicitly set up a dependency on all the cameras you might ask for, and then things work. 

Everything is working as intended, it's just a little surprising.